src/graphics/program-library.js

```diff
@@ -7,6 +7,10 @@ Object.assign(pc, function () {
         this._cache = {};
         this._generators = {};
         this._isClearingCache = false;
+
+        // Shared Sandard Material option structures
+        this.optionsContext = {};
+        this.optionsContextMin = {};

```

Comment:
I'm not sure about the addition of these properties on the pc.ProgramLibrary for a couple of reasons:

1. The src\graphics codebase shouldn't really know about the src\scene layer which is built on top of it.
2. The program library is there to serve more than just the pc.StandardMaterial.

If you can't think of a better place for this, it's fine. But just wanted to mention is somehow feels a little misplaced. Would be good to hear what you think about that.

Reply:
1. The properties objects is part of interface between those 2 layers, and ProgramLibrary is already explicitly depends on  - so IMHO it's not a problem.
2. I absolutely agree.
In general, I'd prefer to have shared (static) objects in standard material namespace.